Sources/Identity/CustomerInfoManager.swift

```diff
@@ -231,21 +235,27 @@ class CustomerInfoManager {
 
     private func sendUpdateIfChanged(customerInfo: CustomerInfo) {
         self.customerInfoCacheLock.perform {
-            guard !self.customerInfoObserversByIdentifier.isEmpty,
-                  self.lastSentCustomerInfo != customerInfo else {
-                      return
-                  }
+            guard self.lastSentCustomerInfo != customerInfo else { return }
 
-            if lastSentCustomerInfo != nil {
+            if self.lastSentCustomerInfo != nil {
                 Logger.debug(Strings.customerInfo.sending_updated_customerinfo_to_delegate)
             } else {
                 Logger.debug(Strings.customerInfo.sending_latest_customerinfo_to_delegate)
             }
 
+            if let previousSubscriptions = self.lastSentCustomerInfo?.activeSubscriptions,
+               previousSubscriptions != customerInfo.activeSubscriptions {
+                self.notificationCenter.post(name: Self.customerInfoChangedActiveSubscriptionsNotification,
+                                             object: customerInfo)
+            }
```

Comment:
I think we should just play it on the safe side here and invalidate cache on _any_ change to the customerInfo. 
What if for some reason the customerInfo just backfilled a bunch of expired subscriptions that would actually change the result of eligibility? 
For example: 

1. Log in as user A for the first time. Eligible for all things. 
2. Restore purchases, there were a bunch of old purchases that this person made as an anonymous user last year. 
3. User is no longer eligible, but there are no active subscriptions, so we're not refreshing cache

Reply:
> I think we should just play it on the safe side here and invalidate cache on any change to the customerInfo.

Makes sense 👍🏻  great feedack.